src/Angor/Client/Pages/Settings.razor

```diff
@@ -471,39 +472,81 @@
             await Refresh(false);
         }
     }
-
+    
     private async Task AddIndexer()
     {
         if (!string.IsNullOrWhiteSpace(newIndexerLink))
         {
             if (!Uri.TryCreate(newIndexerLink, UriKind.Absolute, out Uri? uri))
             {
-                notificationComponent.ShowErrorMessage($"invalid url {newIndexerLink}");
+                notificationComponent.ShowErrorMessage($"Invalid URL: {newIndexerLink}");
                 return;
             }
 
             if (uri.Scheme is not ("http" or "https"))
             {
-                notificationComponent.ShowErrorMessage($"invalid url {newIndexerLink} schema must be http or https");
+                notificationComponent.ShowErrorMessage($"Invalid URL schema. Must be HTTP or HTTPS.");
                 return;
             }
 
-            newIndexerLink = new Uri($"{uri.Scheme}://{uri.Host}").AbsoluteUri.TrimEnd('/');
+            newIndexerLink = $"{uri.Scheme}://{uri.Host}".TrimEnd('/');
+
+            Logger.LogInformation($"Normalized URL: {newIndexerLink}");
 
             if (settingsInfo.Indexers.Any(a => a.Url == newIndexerLink))
             {
-                notificationComponent.ShowErrorMessage($"url exists {newIndexerLink}");
+                notificationComponent.ShowErrorMessage($"URL already exists: {newIndexerLink}");
                 return;
             }
 
-            settingsInfo.Indexers.Add(new SettingsUrl { Url = newIndexerLink, IsPrimary = !settingsInfo.Indexers.Any() });
+            // Add the new indexer temporarily
+            settingsInfo.Indexers.Add(new SettingsUrl
```

Comment:
All this code can go away and we should only call `inderxer.CheckIndexerNetwork` that will check online and also fetch block 0